integration/querydsl/testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -214,10 +230,29 @@ public void testFetchCount() {
         });
     }
 
+    @Test
+    public void testModuloLongOperation() {
+        doInJPA(em -> {
+            BlazeJPAQuery<Tuple> query = new BlazeJPAQuery<>(em, cbf)
+                    .from(person)
+                    .select(person, person)
+                    .where(bitwiseAndLong(person.id, 1L));
+
+            assertNotNull(query.fetchResults());
+        });
+    }
+
+    public static BooleanExpression bitwiseAndLong(NumberExpression<Long> expression, long bit) {
+        NumberExpression<Long> mod = expression.divide(1L << bit).floor().mod(2L);
```

Comment:
According to the Postgresql documentation, `ceil` should return its input type.

However:

```
portal=# select pg_typeof(5::bigint/2::bigint);
 pg_typeof 
-----------
 bigint
(1 row)

portal=# select pg_typeof(floor(5::bigint/2::bigint));
    pg_typeof     
------------------
 double precision
```

This then causes issues with the modulo operator, failing with `Caused by: org.postgresql.util.PSQLException: ERROR: function mod(double precision, bigint) does not exist`.

Should we cast the result of `floor` or just leave it to the user to solve and just add the cast to the test here to fix the issue with Postgres?

Reply:
According to this (https://www.postgresql.org/docs/current/functions-math.html#id-1.5.8.9.6.2.2.9.1.1.1) it will return either numeric or double precision. Since `5/2` is a literal, PostgreSQL will probably infer `double precision` rather than `numeric`. Overall, I think this is something Hibernate 6.x should fix (can you please create an issue?). For the test, I'd say let's just cast `floor`, yes. But let's not change anything in the implementation.

Reply:
Added cast